### ISSUE TITLE
Message after redirect

### ISSUE THREAD

**Issue by iamthestreets** [benedmunds/CodeIgniter-Ion-Auth#864]
Hi,

I have taken the `create_user` function and created a `register` function. When the user successfully registers it will redirect the user to the login page. On the login page I want to display a message using flashdata, but it does not seem to work the same as the `create_user` function.

Here is partial code:
````php
if ($this->form_validation->run() == true && $this->ion_auth->register($username, $password, $email, $agreement))
 {
            // check to see if we are creating the user
            // redirect them back to the admin page
            $this->session->keep_flashdata('message', $this->ion_auth->messages());
			$this->session->keep_flashdata('message_class', 'alert-success');
            redirect("auth/login");
 }
````

I also tried keeping it at `set_flashdata` and neither work. Any idea why I am not getting the message to display on the redirect?

**Comment by avenirer** [comment]
First of all, what do you mean by "taking a function and creating another function"? What did you change? The name? The loaded views?

Also, I think you are using keep_flashdata() wrong... Take another look at the CI manual: http://www.codeigniter.com/user_guide/libraries/sessions.html#flashdata

Partial code? Why not showing us the whole code? :)

**Comment by iamthestreets** [comment]
Create user is meant to be an admin function. I took the code in that function, changed it, and created a new `register` function to allow a user registration.

The only thing I changed is `keep` from `set`. The original code in the `create_user` function was written like this: `$this->session->set_flashdata('message', $this->ion_auth->messages());`. and I added `$this->session->keep_flashdata('message_class', 'alert-success');` to set the bootstrap alert success class.

There was no need to show the whole code. That piece of code I added was good enough to show what was happening at that point. The only thing I could've added that would've helped was the view page code. Which turned out to be the problem. I was getting an error that said I have to check if `null`. `isset` was not enough. So in the login view I added `is_null` like this:
`if(isset($message) || !is_null($message)){` this fixed my problem.
